File: test-suite-spring-java/src/main/java/io/micronaut/openapi/spring/Application.java
package io.micronaut.openapi.spring;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.info.Info;
import org.springframework.boot.Banner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;

@OpenAPIDefinition(
    info = @Info(
        version = "${app.version}"
    )
)
@SpringBootApplication
public class Application {

    public static void main(String[] args) {
        var application = new SpringApplication(Application.class);
        application.setBannerMode(Banner.Mode.OFF);
        application.run(args);
    }
}

File: test-suite-spring-kotlin-ksp/src/main/kotlin/io/micronaut/openapi/spring/Application.kt
package io.micronaut.openapi.spring

import io.swagger.v3.oas.annotations.OpenAPIDefinition
import io.swagger.v3.oas.annotations.info.Info
import org.springframework.boot.Banner
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication

@SpringBootApplication
@OpenAPIDefinition(info = Info(version = "\${app.version}"))
open class Application

fun main(args: Array<String>) {
    val application = SpringApplication(Application::class.java)
    application.setBannerMode(Banner.Mode.OFF)
    application.run(*args)
}
